py/server/deephaven/experimental/iceberg.py

```diff
@@ -250,3 +251,80 @@ def adapter_aws_glue(
     except Exception as e:
         raise DHError(e, "Failed to build Iceberg Catalog Adapter") from e
 
+
+def adapter(
+        properties: Dict[str, str],
+        name: Optional[str] = None
+) -> IcebergCatalogAdapter:
```

Comment:
I think we should compare this directly to https://py.iceberg.apache.org/reference/pyiceberg/catalog/#pyiceberg.catalog.load_catalog, https://github.com/apache/iceberg-python/blob/pyiceberg-0.6.1/pyiceberg/catalog/__init__.py#L185-L200

```python
def load_catalog(name: Optional[str] = None, **properties: Optional[str]) -> Catalog:
```

Should we call this `load_catalog_adapter`, `load_catalog`, or `load_adapter` instead? Should we match the same sort of signature, ie using `**properties`?

In addition, I think from python standpoint, we also need to compare to the wider pyiceberg API to compare / contrast, https://py.iceberg.apache.org/api/

I don't think we necessarily need to dive deep into the details today, but we could potentially add a layer of yaml config in the future like pyiceberg does with `~/.pyiceberg.yaml` (in which case, the config is looked up by name).

Reply:
Based on the order of args, we are not able to do `adapter("name", { "properties": ... })`, which seems like the more natural order?

Reply:
The new call is `IcebergCatalogAdapter.adapter()`, I think that is pretty clear but I'm open to  new name.